### PR TITLE
[WIP] - Update testing and use TestContainers.

### PR DESCRIPTION
**Pull request by dataswifty**
## Description

Our previous CI setup was configured for a dedicated Postgres to be used for the test.  Since moving to GH actions we have moved away from that setup.  We have used testcontainers in other projects successfully, so why not use them for this project?

## AC

Current:The tests that are run in CI require a local Postgresql node to run.

Desired:Switch to our established solution using testcontainers to provide ephemeral resources.

Env:All

Resources:https://www.testcontainers.org

## Source

Link to story/task/bug

## Type

- [x] Bug Fix
- [x] Feature Addition
- [x] Refactor
- [ ] HotFix

## Checklist

- [x] tested locally
- [x] added new dependencies
- [ ] updated the docs
- [x] added a test

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -53,11 +53,15 @@ lazy val hat = project
     libraryDependencies := (buildEnv.value match {
           case BuildEnv.Developement | BuildEnv.Test =>
             libraryDependencies.value ++ Seq(
-                  Library.Play.specs2,
-                  Library.Specs2.core,
-                  Library.Specs2.matcherExtra,
-                  Library.Specs2.mock,
-                  Library.Play.Silhouette.silhouetteTestkit
+                  Library.Play.Silhouette.silhouetteTestkit,
+                  Library.Test.scalatest,
```
Comment (benshaw) [inline build.sbt:57]: Which of these are bought in my testCommon ? Do we need to repeat dependencies that are bought in transitively (reduces version clashes, makes upgrading easier etc)

File: hat/app/org/hatdex/hat/api/controllers/Files.scala [hunk 2]
```diff
@@ -278,33 +261,27 @@ class Files @Inject() (
...
   private def filePermissionsCleaned(
       file: ApiHatFile
     )(implicit user: HatUser,
-      authenticator: HatApiAuthEnvironment#A
-    ): ApiHatFile = {
-    if (WithRole.isAuthorized(user, authenticator, Owner())) {
+      authenticator: HatApiAuthEnvironment#A): ApiHatFile =
+    if (WithRole.isAuthorized(user, authenticator, Owner()))
       file
-    } else {
```
Comment (benshaw) [inline hat/app/org/hatdex/hat/api/controllers/Files.scala:293]: Non blocking: i thought our styling rules where to include brackets? Not related to this PR but we should confirm whether we want or dont want brackets and ensure the tools are applying the correct rules

File: hat/test/org/hatdex/hat/api/HATTestContext.scala [hunk 3]
```diff
@@ -47,51 +47,155 @@ import org.hatdex.hat.phata.models.MailTokenUser
...
 import play.api.{ Application, Configuration, Logger }
 import play.cache.NamedCacheImpl
+import com.dimafeng.testcontainers.{ ForAllTestContainer, PostgreSQLContainer }
 
 import scala.concurrent.duration._
 import scala.concurrent.{ Await, Future }
+import org.scalatestplus.mockito.MockitoSugar
 
-trait HATTestContext extends Scope with Mockito {
+trait HATTestContext extends MockitoSugar {
```
Comment (benshaw) [inline hat/test/org/hatdex/hat/api/HATTestContext.scala:64]: I was under the impression that we were using scalamock not Mockito for mocking. Is this going to clash with the scala test common if Mockito is used here ?

File: build.sbt (hunk 1 above)
Comment (dataswifty) [reply build.sbt:54]: Good question, I will need to investigate.

File: hat/app/org/hatdex/hat/api/controllers/Files.scala (hunk 2 above)
Comment (dataswifty) [reply hat/app/org/hatdex/hat/api/controllers/Files.scala:293]: Let me verify the scalafmt rules that my pre-commit hook is using.

Comment (benshaw) [reply hat/app/org/hatdex/hat/api/controllers/Files.scala:293]: ive since noticed it other places, its ok

**Review by benshaw — APPROVED** [review]

**Pull request merged**